docs/reference/inference/service-cohere.asciidoc

```diff
@@ -0,0 +1,183 @@
+
+[discrete]
+[[infer-service-cohere-api-path-params]]
+==== {api-path-parms-title}
+
+`<inference_id>`::
+(Required, string)
+include::inference-shared.asciidoc[tag=inference-id]
+
+`<task_type>`::
+(Required, string)
+include::inference-shared.asciidoc[tag=task-type]
++
+--
+Available task types:
+* `completion`,
+* `rerank`,
+* `text_embedding`.
+--
+
+[discrete]
+[[infer-service-cohere-api-request-body]]
+==== {api-request-body-title}
+
+`service`::
+(Required, string)
+The type of service supported for the specified task type. In this case, 
+`cohere`.
+
+`service_settings`::
+(Required, object)
+include::inference-shared.asciidoc[tag=service-settings]
++
+--
+These settings are specific to the `service` you specified which is `cohere` in this case.
+--
+
+`api_key`:::
+(Required, string)
+A valid API key of your Cohere account.
+You can find your Cohere API keys or you can create a new one
+https://dashboard.cohere.com/api-keys[on the API keys settings page].
++
+--
+include::inference-shared.asciidoc[tag=api-key-admonition]
+--
+
+`model_id`::
+(Optional, string)
+The name of the model to use for the {infer} task.
+To review the available `rerank` models, refer to the
+https://docs.cohere.com/reference/rerank-1[Cohere docs].
+
+To review the available `text_embedding` models, refer to the
+https://docs.cohere.com/reference/embed[Cohere docs].
+The default value for `text_embedding` is `embed-english-v2.0`.
+
+`rate_limit`:::
+(Optional, object)
+By default, the `cohere` service sets the number of requests allowed per minute to `10000`.
+This value is the same for all task types.
+This helps to minimize the number of rate limit errors returned from Cohere.
+To modify this, set the `requests_per_minute` setting of this object in your service settings:
++
+--
+include::inference-shared.asciidoc[tag=request-per-minute-example]
+--
++
+--
+More information about Cohere's rate limits can be found in https://docs.cohere.com/docs/going-live#production-key-specifications[Cohere's production key docs].
+--
++
+.`service_settings` for the `text_embedding` task type
+[%collapsible%closed]
+=====
+`embedding_type`:::
+(Optional, string)
+Only for `text_embedding`.
+Specifies the types of embeddings you want to get back.
+Defaults to `float`.
+Valid values are:
+* `byte`: use it for signed int8 embeddings (this is a synonym of `int8`).
+* `float`: use it for the default float embeddings.
+* `int8`: use it for signed int8 embeddings.
+=====
+
+
+`task_settings`::
+(Optional, object)
+include::inference-shared.asciidoc[tag=task-settings]
++
+.`task_settings` for the `rerank` task type
+[%collapsible%closed]
+=====
+`return_documents`::
+(Optional, boolean)
+Specify whether to return doc text within the results.
+
+`top_n`::
+(Optional, integer)
+The number of most relevant documents to return, defaults to the number of the documents.
+=====
++
+.`task_settings` for the `text_embedding` task type
+[%collapsible%closed]
```

Comment:
This section doesn't seem to be rendering correctly
![image](https://github.com/elastic/elasticsearch/assets/56361221/da2e2a5f-a210-4426-b173-63e997e3f5ce)

Reply:
I think this is still rendering like the image I showed. There shouldn't be a `+` showing up right?

Reply:
The preview hasn't been updated yet, the docs-build-pr CI is still working. I have a feeling that I've already fixed it, just the preview still shows the previous state. Not entirely sure though.

Reply:
Yup, it seems okay now.